object-store/src/main/java/org/hypertrace/config/objectstore/DefaultObjectStore.java

```diff
@@ -91,12 +101,24 @@ protected DefaultObjectStore(
           if (response.hasPrevConfig()) {
             configChangeEventGenerator.sendUpdateNotification(
                 context,
-                upsertedObject.getData().getClass().getName(),
-                response.getPrevConfig(),
-                response.getConfig());
+                this.buildClassNameForChangeEvent(upsertedObject.getData()),
+                this.buildDataFromValue(response.getPrevConfig())
+                    .map(this::buildValueForChangeEvent)
+                    .orElseGet(
+                        () -> {
+                          log.error(
+                              "Unable to convert previousValue back to data for change event. Falling back to raw value {}",
+                              response.getPrevConfig());
+                          return response.getPrevConfig();
```

Comment:
We should not have this special case. Please note consumer will not be able to handle it.

Reply:
Consumer may or may not be able to, but at least we give them the option. This maps to the existing behavior (where we put in prevConfig as is, and don't check if it's transformable), but certainly open to other ideas (see https://github.com/hypertrace/config-service/pull/77#discussion_r739336351 for various options I came up with)